src/server/listener.cc

```diff
@@ -444,6 +444,13 @@ void Listener::setupSSLAuth(const std::string &ca_file,
                      );
 }
 
+bool Listener::setSessionIdContext(const std::string &s_id_ctx)
+{
+    return SSL_CTX_set_session_id_context(GetSSLContext(ssl_ctx_),
+                                         (const unsigned char*)s_id_ctx.c_str(),
```

Comment:
I've fixed the casts. I'll have to do some digging on the Get part, doesn't seem to be an equivalent Get function, but there seems to be other ways of getting it. I'll look through your existing tests and see if I can figure out how to test this in a meaningful way. If I get a Get function implemented then I can at least do a get-matches-set test, but I'll try to reproduce the original issue and create a test for that.

Reply:
Ok, so after a lot of digging around in openssl it seems that they really don't want you to poke around in the SSL_CTX as it is an "opaque structure". I can get around it by adding a callback to get all new sessions that are established and use ```const unsigned char *SSL_SESSION_get0_id_context(const SSL_SESSION *s, unsigned int *len)``` but it seems that this should be considered a write only option.  
  
I could also just store the value passed into my set function and return that if present, but not sure that it will help anyone.

Reply:
As far as SSL goes, I don't know.  I've never looked under the hood there, nor coded to their library.  SSL support in Pistache was added by another contributor.  My only concern with this PR is that we should not use C-style casts for getting anything, including the "(const char*)" and "(unsigned int)" casts that are in your PR.  Please convert them to C++ style casts.

As for dealing with SSL; Do what you must.  Please add a unit test - even if you cannot verify that the session ID was set correctly, verify that your call to set it does not crash.

Thank you very much for your contribution and patience.

Reply:
Hi again, I've fixed the casts and I'll submit a simple unit test, frustrating to not being able to reproduce. I think I'll have time this weekend to fix it.